core/src/com/unciv/logic/automation/civilization/ReligionAutomation.kt

```diff
@@ -13,7 +15,181 @@ import kotlin.math.min
 import kotlin.math.pow
 import kotlin.random.Random
 
-object ChooseBeliefsAutomation {
+object ReligionAutomation {
+
+    // region faith spending
+
+    fun spendFaithOnReligion(civInfo: CivilizationInfo) {
+        if (civInfo.cities.isEmpty()) return
+
+        // Save for great prophet
+        if (civInfo.religionManager.religionState != ReligionState.EnhancedReligion
+            && (civInfo.religionManager.remainingFoundableReligions() != 0 || civInfo.religionManager.religionState > ReligionState.Pantheon)
+        ) {
+            buyGreatProphetInAnyCity(civInfo)
+            return
+        }
+
+        // We don't have a religion and no more change of getting it :(
+        if (civInfo.religionManager.religionState <= ReligionState.Pantheon) {
+            tryBuyAnyReligiousBuilding(civInfo)
+            // Todo: buy Great People post industrial era
+            return
+        }
+
+        // If we don't have majority in all our own cities, build missionaries and inquisitors to solve this
+        val citiesWithoutOurReligion = civInfo.cities.filter { it.religion.getMajorityReligion() != civInfo.religionManager.religion!! }
+        // The original had a cap at 4 missionaries total, but 1/4 * the number of cities should be more appropriate imo
+        if (citiesWithoutOurReligion.count() >
+            4 * civInfo.getCivUnits().count { it.canDoReligiousAction(Constants.spreadReligion) || it.canDoReligiousAction(Constants.removeHeresy) }
+        ) {
+            val (city, pressureDifference) = citiesWithoutOurReligion.map { city ->
+                city to city.religion.getPressureDeficit(civInfo.religionManager.religion?.name)
+            }.maxBy { it.second }
+            if (pressureDifference >= 3000)
```

Comment:
This 3000 number appears in at least two places (here and in the inquisitor AI code). Is there a significance to it or is it an arbitrary value?